File: extractInfoNCAA.pl
#!/usr/bin/perl
use strict;
use Data::Dumper;
use HTML::TableExtract;

use Text::CSV;

my $csv = Text::CSV->new ( { binary => 1 } )  # should set binary attribute.
                 or die "Cannot use CSV: ".Text::CSV->error_diag ();


my $path = "/home/paul/Downloads/";
my $inFileNameSportCodes = "NCAA-SportsCodes.csv";
my $inFileNameSportInfo = "NCAA-SportsInfo-";
my $outFileName = "NCAA-SportsInfo.csv";


mainProcess($path, $inFileNameSportCodes, $inFileNameSportInfo, $outFileName);

sub mainProcess{
    my $path = shift @_;
    my $inNm = shift @_;
    my $inNmPartial = shift @_;
    my $outNm = shift @_;

    my $fullNm = $path . $inNm;
    open (my $inFH, "$fullNm") ||die "can't open $fullNm";

    my $fullNm = $path . $outNm;
    open (my $outFH, ">", "$fullNm") ||die "can't open $fullNm";

    while (<$inFH>){
        chomp;
        my($sportCode,$sportName) = split(/,/,$_);
        print "processing the sportCode:$sportCode\t$sportName\n";
        processFile($path, $sportCode, $inNmPartial, $outFH);
        print "one down\n";
    }
    close $inFH;
    close $outFH;


}

sub processFile{
    my $path = shift @_;
    my $sportCode = shift @_;
    my $inNmPartial = shift @_;
    my $outFH = shift @_;

    my $fullNm = $path . $inNmPartial . $sportCode . ".html";
    open (my $inFH, "$fullNm") ||die "can't open $fullNm";

    my $content = "";
    while (<$inFH>){
        $content .= $_;
    }
    close $inFH;

    $content =~ s/\n//g;
    $content =~ s/\t//g;

#    for ($iDepth = 1; $iDepth <= 1; $iDepth++) {
    for (my $iDepth = 1) {

#         for ($iCount = 0; $iCount <= 15; $iCount++) {
         for (my $iCount = 0) {

            my $te = ();
            $te = HTML::TableExtract->new( depth => $iDepth, count => $iCount );
            $te->parse( $content );

            foreach my $ts ( $te->tables() )
            {
                foreach my $row ( $ts->rows() ){
                    #print "depth:$iDepth  count:$iCount\t" .  join(", ", @$row) . "\n";

                    s/^\s+|\s+$//g for (@$row);   ## trim ALL the items
                    if (length($$row[3])){
                        ##print "this is reclass:$$row[3]\n";
                        $$row[2] = $$row[3];      ## update the real division with the RECLASS
                    }

                    splice @$row, 3, 1;           ## remove the RECLASS item

                    print $outFH "$sportCode,";
                    #print Dumper($row);
                    $csv->print ($outFH, $_) for $row ;
                    print $outFH "\n";
                #print $fhOUT "depth:$iDepth,count:$iCount," .  join(", ", @$row) . "\n";
                }
            }

         }

     }
     print "done process $sportCode\n";

}

File: extractInfoNCAA.php
<?php

include 'simple_html_dom.php';



### FILES and URLS
$urlSportCodes = 'http://web1.ncaa.org/onlineDir/exec2/sponsorship';
$fNmSportCodesHTML = '/home/paul/Downloads/NCAA-SportsCodes.html';
$fNmSportCodesCSV = '/home/paul/Downloads/NCAA-SportsCodes.csv';
$fNmSportInfoCSV = '/home/paul/Downloads/NCAA-SportsInfo.csv';


########## ADJUST THESE AS NEEDED!  - EG - DONT RETRIEVE WEB DATA UNLESS NECESSARY
retrieveSportCodesFromWeb($urlSportCodes,$fNmSportCodesHTML);
generateSportCodeCSV($fNmSportCodesHTML, $fNmSportCodesCSV);

//retrieveSportInfoFromWeb($fNmSportCodesCSV);



// THIS FOLLOWING FUNCTION DOES NOT WORK.. USE THE PERL SCRIPT TO GET THE FINAL FILE!
//generateSportInfoCSV($fNmSportCodesCSV,$fNmSportInfoCSV);



############## START FUNCTIONS  #####################################

##################################### GET SPORTS CODE HTML #####
function retrieveSportCodesFromWeb($url,$fNm) {
    getURLWriteLocal($url,$fNm);
}


##################################### EXTRACT SPORTS CODES #####
function generateSportCodeCSV($inNm, $outNm){
        #### OPEN OUTPUT FILE
        $sportCodeFH = fopen($outNm, "w") or die("Unable to open file 1!");

        #### READ INPUT FILE
        print "going to read SportCode HTML from file:$inNm\n";
        $htmlNBR1 = file_get_html($inNm);

        foreach( $htmlNBR1->find('select[name="sport"] option') as $option ){
            $sportCode = $option->value;
            $sportName = $option->plaintext;
//            $txt = "$sportCode,$sportName\n";
            $cellData = array($sportCode,$sportName);
            fputcsv($sportCodeFH, $cellData);
//            fwrite($sportCodeFH, $txt);
        }
        fclose($sportCodeFH);
}

##################################### GET SPORTS INFO HTML #####
function retrieveSportInfoFromWeb($fileNM){
        $handle = fopen($fileNM,'r');
        while ( ($data = fgetcsv($handle) ) !== FALSE ) {
                $sportCode = $data[0];
                print "sportCode:$sportCode\n";
                $url = "http://web1.ncaa.org/onlineDir/exec2/sponsorship?sortOrder=0&division=All&sport=" . $sportCode;
                $localFileName = "/home/paul/Downloads/NCAA-SportsInfo-$sportCode.html";
                getURLWriteLocal($url,$localFileName);
        }
        fclose($handle);
}



#####################################  EXTRACT SPORTS INFO DRIVER #####
function generateSportInfoCSV($fileNM, $outNm){

        $sportInfoFH = fopen($outNm, "w") or die("Unable to open file 7!");

        $handle = fopen($fileNM,'r');
        while ( ($data = fgetcsv($handle) ) !== FALSE ) {
                $sportCode = $data[0];
                print "sportCode:$sportCode\n";
                $localFileName = "/home/paul/Downloads/NCAA-SportsInfo-$sportCode.html";
                processSportCode($sportCode, $localFileName, $sportInfoFH);
        }
}



#####################################  EXTRACT SPORTS INFO #####
function processSportCode($sportCode, $inNm, $outFH) {

        print "reading this file to get sports info:$inNm\n";
        $html = file_get_html($inNm);
        $table = $html->find('table', 1);
        $rowData = array();


        $items = array_slice($html->find('tr'), 1);

        foreach($items as $item){
            print "THIS IS AN ITEM $item\n\n\n\n\n\n";
        }
        die;

        foreach($table->find('tr') as $row) {
            print "found row\n";
            $myRowAsString = $row;
//            print "myRowAsString:$myRowAsString\n\n\n\n\n";

            // initialize array to store the cell data from each row
            $cellData = array();

            foreach($row->find('td') as $cell) {
                    $myRowAsString = $cell;
//                    print "myRowAsString:$myRowAsString\n";
//                    print "this is plaintext:$cell->plaintext and this is inner:$cell->innertext\n";
                    //die;

                    //foreach($row->find('td') as $cell) {
                    // push the cell's text to the array
                    //print "found td as cell\n";
                    //print "this is plaintext:$cell->plaintext and this is inner:$cell->innertext\n";
                    $cellData[] = $cell->plaintext;
            }
//            echo "$cellData[0]--\t$cellData[1]--\t$cellData[2]--\t$cellData[3]--\t$cellData[4]--\t$cellData[5]\n";
            $txt = "$sportCode\t";
            fwrite($outFH, $txt);
            fputcsv($outFH, $cellData);

        }
}




#####################################  GET WEB PAGE AND WRITE TO LOCAL DISK #####
function getURLWriteLocal($url,$localNM) {
        print "Getting data from this URL:$url\n";
        print "HTML will be written to:$localNM\n";

        $html = file_get_contents($url);
        ###############echo "$html";
        $FH = fopen($localNM, "w") or die("Unable to open file to hold raw HTML!");
        fwrite($FH, $html);
        fclose($FH);
}



?>